src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -53798,11 +53771,315 @@ static void F(A a, B b)
+                //         var t3 = ((S?, B))(new S() { F = 3 }, new A()); // 4, 5
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "((S?, B))(new S() { F = 3 }, new A())").WithArguments("(S?, B?)", "(S?, B)").WithLocation(28, 18),
+                // (28,47): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         var t3 = ((S?, B))(new S() { F = 3 }, new A()); // 4, 5
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "new A()").WithLocation(28, 47),
+                // (30,9): warning CS8602: Possible dereference of a null reference.
+                //         t3.Item2.ToString(); // 6
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t3.Item2").WithLocation(30, 9));
+        }
+
+        [Fact]
+        [WorkItem(32599, "https://github.com/dotnet/roslyn/issues/32599")]
+        [WorkItem(32600, "https://github.com/dotnet/roslyn/issues/32600")]
+        public void Conversions_TupleConversions_11()
+        {
+            var source =
+@"class A
+{
+    public static implicit operator B?(A a) => null;
+}
+class B
+{
+}
+struct S
+{
+    internal object? F;
+}
+class Program
+{
+    static void F1()
+    {
+        (A, S) t1 = (new A(), new S() { F = 1 });
+        (B, S?) u1 = t1; // 1
+        u1.Item1.ToString(); // 2
+        u1.Item2.Value.F.ToString();
+    }
+    static void F2()
+    {
+        (A, S) t2 = (new A(), new S() { F = 2 });
+        var u2 = ((B, S?))t2; // 3, 4
+        u2.Item1.ToString(); // 5
+        u2.Item2.Value.F.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/32599: Track state across tuple element conversions.
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(32599, "https://github.com/dotnet/roslyn/issues/32599")]
+        [WorkItem(32600, "https://github.com/dotnet/roslyn/issues/32600")]
+        public void Conversions_TupleConversions_12()
+        {
+            var source =
+@"#pragma warning disable 0649
+class A
+{
+    public static implicit operator B?(A a) => null;
+}
+class B
+{
+}
+struct S
+{
+    internal object? F;
+}
+class Program
+{
+    static void F1()
+    {
+        (int, (A, S)) t1 = (1, (new A(), new S() { F = 1 }));
+        (object x, (B, S?) y) u1 = t1; // 1
+        u1.y.Item1.ToString(); // 2
+        u1.y.Item2.Value.F.ToString();
+    }
+    static void F2()
+    {
+        (int, (A, S)) t2 = (2, (new A(), new S() { F = 2 }));
+        var u2 = ((object x, (B, S?) y))t2; // 3
+        u2.y.Item1.ToString(); // 4
+        u2.y.Item2.Value.F.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/32599: Track state across tuple element conversions.
```

Comment:
>    // https://github.com/dotnet/roslyn/issues/32599: Track state across tuple element conversions. [](start = 8, length = 99)

Comment could be removed?

Reply:
We should report warnings marked above. I have a fix for this case in the next PR.

---
In reply to: [268849176](https://github.com/dotnet/roslyn/pull/34303#discussion_r268849176) [](ancestors = 268849176)